### PR TITLE
make logs more useful, fix octokit request

### PR DESCRIPTION
**Pull request by NovemberTang**
## What does this change?

We are submitting the full repo name to github. We shouldn't be doing this
Also make repocop logging more useful

## Why?

Passing in the full name into the name field means we are searching for the wrong repo and return a 404. ie `guardian/guardian%2Fcapi` rather than `guardian/capi`

## How has it been verified?

Deployed to code, octokit is able to find the correct repo

### CODE REVIEW

File: packages/interactive-monitor/src/index.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ export const handler: SNSHandler = async (event) => {
 			`Expected exactly one record, but got ${event.Records.length}`,
 		);
 	} else {
-		const repo = event.Records[0]!.Sns.Message;
+		const repo = event.Records[0]!.Sns.Message.split('/')[1]!;
```
Comment (NovemberTang) [inline packages/interactive-monitor/src/index.ts:29]: the non null assertion is a bit controversial but i think we should trust that repocop has sent what it's promised to send. If not, we should throw an error anyway

File: packages/interactive-monitor/src/index.ts [hunk 2]
```diff
@@ -21,12 +21,15 @@ async function isFromInteractiveTemplate(
 
 export const handler: SNSHandler = async (event) => {
 	const config = getConfig();
-	if (event.Records.length !== 1) {
+	if (
+		event.Records.length !== 1 ||
```
Comment (akash1810) [inline packages/interactive-monitor/src/index.ts:25]: Are we always guaranteed to have exactly one record? Might it be simpler to loop over them?

```ts
event.Records.forEach((record) => {
		const message = record.Sns.Message;
		const fullRepoName = message.split('/');
		if (fullRepoName.length === 2) {
			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- we've checked the length above
			const repo = fullRepoName[1]!;
			
		}
	});
```

**Review by akash1810 — COMMENTED** [review]
> Passing in the full name into the name field means we are searching for the wrong repo and return a 404. ie `guardian/guardian%2Fcapi` rather than `guardian/capi`

Should we not fix this at source instead, so that we always produce the right message?

File: packages/interactive-monitor/src/index.ts (hunk 2 above)
Comment (NovemberTang) [reply packages/interactive-monitor/src/index.ts]: ok i think both of these are fixed now. It's unlikely we'll be handed multiple records but can't hurt to cover our bases. and it looks nicer.

File: packages/interactive-monitor/src/index.ts [hunk 3]
```diff
@@ -19,30 +21,34 @@ async function isFromInteractiveTemplate(
...
-	} else {
-		const repo = event.Records[0]!.Sns.Message;
-		const owner = 'guardian';
-		console.log('received repo', repo);
-		const githubAppConfig: GitHubAppConfig = await getGitHubAppConfig();
-		const octokit: Octokit = await getGithubClient(githubAppConfig);
+async function applyTopics(repo: string, owner: string, octokit: Octokit) {
+	const topics = (await octokit.rest.repos.getAllTopics({ owner, repo })).data
+		.names;
+	console.log(`${repo}is from interactive template`);
```
Comment (akash1810) [inline packages/interactive-monitor/src/index.ts:27]: Missing a space?

```ts
	console.log(`${repo} is from interactive template`);
```

File: packages/repocop/src/remediations/repository-06-topic-monitor-interactive.ts [hunk 4]
```diff
@@ -7,3 +15,51 @@ export function findPotentialInteractives(
 		.filter((repo) => !repo.topics)
 		.map((repo) => repo.full_name);
 }
+
+export function createBatchEntry(string: string): PublishBatchRequestEntry {
```
Comment (akash1810) [inline packages/repocop/src/remediations/repository-06-topic-monitor-interactive.ts:19]: minor: Could we give the argument a more intention revealing name? Maybe `fullRepoName`?

**Review by akash1810 — APPROVED** [review]

**Review by tjsilver — APPROVED** [review]
Looks good, and like we'll be able to move some of the functions into 'common' when the production-monitor comes online!

**Pull request merged**